src/components/buttons/_buttons.scss

```diff
@@ -272,6 +292,21 @@ $largeButtonSize: componentSize(l);
       &--l {
         margin-left: -3px;
       }
+
+      &--right-aligned {
+        margin-right: -2px;
+        margin-left: spacing(xs);
+      }
+    }
+
+    .sg-button__icon--right-aligned.sg-button__icon--s {
```

Comment:
not sure why, but `&--modifier#{&}--modifier-2` doesn't work here - I had to do it this way :(